pyrobosim/test/system/test_system.py

```diff
@@ -142,3 +142,39 @@ def test_open_close(self):
             # Open the location and verify that it's open.
             window.on_open_click()
             assert location.is_open
+
+    def cancel_helper(self, nav_query):
+        """
+        Helper function to test cancel UI action.
+
+        :param nav_query: Query for navigation goal.
+        :type nav_query: str
+        """
+        window = self.app.main_window
+        robot = window.get_current_robot()
+
+        window.goal_textbox.setText(nav_query)
+        window.on_navigate_click()
+
+        while not robot.executing_nav:
+            time.sleep(0.2)
+        if robot.executing_nav:
+            window.on_cancel_action_click()
+
+        assert robot.canceling_execution == True
+
+    @pytest.mark.dependency(name="test_cancel_action")
+    def test_cancel_action(self):
```

Comment:
Because all these tests have to run sequentially, not in parallel, you need to additionally include that `depends=` argument as in the other tests.

```suggestion
    @pytest.mark.dependency(name="test_cancel_action", depends=["test_open_close"])
    def test_cancel_action(self):
```